### PR TITLE
FIX: Automation `topic_tags_changed` should not fire on PMS

### PR DESCRIPTION
**Pull request by Grubba27**
automation on topic tags should not fire on PMS

### CODE REVIEW

File: spec/integration/automation_spec.rb [hunk 1]
```diff
@@ -84,5 +86,24 @@
...
+
+    it "should not trigger notifications for PMs" do
+      automation.upsert_field!(
+        "watching_categories",
+        "categories",
+        { "value" => [nil] },
+        target: "trigger",
+      )
+
+      topic =
```
Comment (markvanlan) [inline spec/integration/automation_spec.rb:98]: Can you use a `pm = Fabricate(:private_message_topic)` to create the topic, rather than TopicCreator?

**Comment by jjaffeux** [comment]
Are we sure we never want this to happen? Should it be an option which would default to not fire on PMs?

**Comment by Grubba27** [comment]
I've added @jjaffeux this:

Screenshot 2024-10-08 at 09 14 30

It is acceptable to be in the script part, but if we continue migrating rules to automation in the future, it would be nice to move this `trigger with PMs` to the automation side(trigger).

**Comment by Grubba27** [comment]
closed in favor of https://github.com/discourse/discourse/pull/29122

**Pull request closed without merging**